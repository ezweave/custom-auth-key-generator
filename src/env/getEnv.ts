import { Config, parseConfig } from '../config';

export type Env = Config;

export const getEnv = (required = true): Env => parseConfig(process.env, required);
