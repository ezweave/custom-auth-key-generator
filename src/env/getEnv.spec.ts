import { getEnv } from './getEnv';

describe('getEnv', () => {
    it('gets the environment variables', () => {
        expect(getEnv()).toMatchSnapshot();
    });

    it('throws errors if required env fields are not present', () => {
        process.env.PUBLIC_KEY = undefined;
        process.env.PRIVATE_KEY = undefined;
        expect(() => getEnv()).toThrowErrorMatchingSnapshot();
    });

    it('doesnt throw errors if required is turned off', () => {
        process.env.PUBLIC_KEY = undefined;
        process.env.PRIVATE_KEY = undefined;
        expect(getEnv(false)).toMatchSnapshot();
    });
});
