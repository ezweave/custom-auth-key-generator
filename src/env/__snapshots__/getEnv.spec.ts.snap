// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEnv doesnt throw errors if required is turned off 1`] = `
Object {
  "headerPrefix": "Somekind of Auth",
  "privateKey": "undefined",
  "publicKey": "undefined",
}
`;

exports[`getEnv gets the environment variables 1`] = `
Object {
  "headerPrefix": "Somekind of Auth",
  "privateKey": "PRIVATE_KEY",
  "publicKey": "PUBLIC_KEY",
}
`;

exports[`getEnv throws errors if required env fields are not present 1`] = `
"Missing value for required field 'privateKey'/'PRIVATE_KEY'
Missing value for required field 'publicKey'/'PUBLIC_KEY'"
`;
